.github/workflows/lint-test-build-push.yaml

```
@@ -0,0 +1,207 @@
+
+    steps:
+      # Create a short SHA-tag to tag built images
+      - name: Create SHA Container Tag
+        id: sha_tag
+        run: |
+          tag=$(cut -c 1-7 <<< $GITHUB_SHA)
+          echo "::set-output name=tag::$tag"
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GHCR_TOKEN  }}
+
+      # Set up a caching directory for image layers. According to the docker
+      # documentation, it's recommended to use a SHA-based key to get the
+      # greatest change of finding the most relevant cached layer. We fall
+      # down to more generic containers by then matching by GitHub branch,
+      # to use cache generated earlier in the same branch, and finally to
+      # the latest cache in general. The `v0` is purely a cache version
+      # indicator that can be incremented manually if we want to invalidate
+      # old caches completely.
+      - name: Cache Image Layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-v0-buildx-${{ github.ref }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-v0-buildx-${{ github.ref }}-
+            ${{ runner.os }}-v0-buildx-
+
+      # Build the image we need for testing/linting the current codebase,
+      # without pushing the image to the GHCR. Instead, we load it into
+      # the runner's docker environment so we can run it later. The
+      # target of this build is the `venv` stage of the Dockerfile, as we
+      # don't want to include the final production entry point stage.
+      #
+      # This build caches to our GitHub Actions cache and uses that cache
+      # during the build process as well. If no GitHub Actions cache was
+      # available, it will use the latest intermediate images pushed to
+      # the GHCR as a cache source.
+      - name: Build image for linting and testing
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./docker/Dockerfile
+          push: false
+          load: true
+          target: venv
+          cache-from: |
+            type=local,src=/tmp/.buildx-cache
+            ghcr.io/python-discord/snekbox-base:latest
+            ghcr.io/python-discord/snekbox-venv:latest
+          cache-to: type=local,dest=/tmp/.buildx-cache,mode=max
+          tags: ghcr.io/python-discord/snekbox-venv:${{ steps.sha_tag.outputs.tag }}
+
+      - name: Start Container
+        run: "docker run \
+          --tty \
+          --detach \
+          --name snekbox_test \
+          --privileged \
+          --hostname pdsnk-dev \
+          -e PYTHONDONTWRITEBYTECODE=1 \
+          -e PIPENV_PIPFILE='/snekbox/Pipfile\' \
```

What's up with this weird escape? And why does the pipeline work with it?

```suggestion
          -e PIPENV_PIPFILE='/snekbox/Pipfile' \
```